tests/testthat/test-summarize.R

```diff
@@ -75,8 +77,30 @@ with_mock_crunch({
                 group_by(gear, add=TRUE) %>%
                 summarize(hp=mean(hp), sd_hp=sd(hp), count=n())
         expect_is(tbl7, "tbl_df")
-        expect_identical(dim(tbl7), c(6L, 5L))
-        expect_identical(names(tbl7), c("vs", "gear", "hp", "sd_hp", "count"))
+        expect_identical(dim(tbl7), c(6L, 6L))
+        expect_identical(names(tbl7), c("vs", "gear", "is_missing", "hp", "sd_hp", "count"))
         expect_equal(as.numeric(filter(tbl7, vs == 1 & gear == 4)$count), 2)
     })
+    
+    test_that("unweighted_counts", {
+
+        tbl8 <- ds %>% 
+            summarize(n = unweighted_n())
+        expect_equal(dim(tbl8), c(1,1))
+        expect_equal(names(tbl8), "n")
+        
+        tbl9 <- ds %>% 
+            summarize(mean = mean(hp), n = unweighted_n())
+        expect_equal(dim(tbl9), c(1,2))
+        expect_equal(names(tbl9), c("mean", "n"))
+        
+        tbl10 <- ds %>% 
+            group_by(cyl, gear) %>% 
+            summarize(mean = mean(hp), n = unweighted_n())
+        expect_equal(dim(tbl10), c(9, 5))
+        expect_equal(names(tbl10), c("cyl", "gear", "is_missing", "mean", "n"))
+        
```

Comment:
So after the extra gymnastics to make sure that if you ask for unweighted_n more than once, it doesn't break, we should have a test for that :)